modules/secrets-store-vault.adoc

```diff
@@ -0,0 +1,384 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="secrets-store-vault_{context}"]
+= Mounting secrets from HashiCorp Vault
+
+You can use the {secrets-store-operator} to mount secrets from HashiCorp Vault to a CSI volume in {product-title}.
+
+// TODO: Is there a requirement for particular provider? https://issues.redhat.com/browse/OSDOCS-10136 steps say Azure, is this true?
+// To mount secrets from HashiCorp Vault, your cluster must be installed on Microsoft Azure.
+
+.Prerequisites
+
+// TODO
+// * Your cluster is installed on Azure.
+* You have installed the {secrets-store-operator}. See _Installing the {secrets-store-driver}_ for instructions.
+* You have installed Helm.
+* You have installed HashiCorp Vault (`vault`).
+* You have configured HashiCorp Vault to store the required secrets.
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Add the HashiCorp Helm repository by running the following command:
++
+[source,terminal]
+----
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
+----
+
+. Update all repositories to ensure that Helm is aware of the latest versions by running the following command:
++
+[source,terminal]
+----
+$ helm repo update
+----
+
+. Create a new project for Vault by running the following command:
++
+[source,terminal]
+----
+$ oc new-project vault
+----
+
+. Install the HashiCorp Vault provider:
++
+////
+// TODO: Confirm if we still need this step to label the ns if we're doing a deployment instead of a pod
```

Comment:
@ropatil010 In the Google doc, I asked why these labelling steps were necessary, since they aren't in the AWS/Azure procedures. 

You mentioned that they were necessary if users are creating a pod instead of a deployment. But since we updated the later step to be a deployment, does this mean that we don't need this step after all?

Reply:
From my understanding if we are using ns vault, we need to add the labels and scc policy. If we do not add the labels/scc then vault provider ds wont reach to running state. We get error as 
"Warning  FailedCreate  2m28s (x7 over 4m54s)   daemonset-controller  Error creating: pods "vault-csi-provider-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, spec.volumes[0]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, provider restricted-v2: .containers[0].privileged: Invalid value: true: Privileged containers are not allowed, provider restricted-v2: .containers[1].runAsUser: Invalid value: 100: must be in the ranges: [1000720000, 1000729999], provider "restricted": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]"

Reply:
Okay as long as it's okay to turn off PSA label syncing. I'll add it back in, thanks!